layouts/default.vue

```diff
@@ -92,10 +94,10 @@
       >
         <!-- Site Header -->
 
-        <div class="flex h-12 align-middle">
-          <sidebar-toggle @click="toggleSidebar" />
-          <breadcrumb-links class="flex-grow" />
-          <theme-switcher class="inline-block" />
+        <div class="flex h-12 items-center gap-1 px-2 sm:pl-8">
+          <SidebarToggle class="sm:hidden" @click="toggleSidebar" />
+          <BreadcrumbLinks class="flex-grow" />
+          <ThemeSwitcher />
```

Comment:
Layout of these elements is now defined where they are used here in this parent component instead of coupling the layout to the components themselves by having margin classes within the components